Library/Homebrew/test/test_diagnostic.rb

```diff
@@ -141,4 +142,80 @@ def test_check_user_curlrc
         @checks.check_user_curlrc
     end
   end
+
+  def test_check_for_unsupported_curl_vars
+    MacOS.stubs(:version).returns :yosemite
+    ENV["SSL_CERT_DIR"] = "/some/path"
+
+    assert_match "SSL_CERT_DIR support was removed from Apple's curl.",
+      @checks.check_for_unsupported_curl_vars
+  end
+
+  def test_check_for_config_scripts
+    mktmpdir do |path|
+      FileUtils.touch "#{path}/foo-config"
+      ENV["PATH"] = "#{path}#{File::PATH_SEPARATOR}#{ENV["PATH"]}"
+
+      assert_match %("config" scripts exist),
+        @checks.check_for_config_scripts
+    end
+  end
+
+  def test_check_DYLD_vars
+    ENV["DYLD_INSERT_LIBRARIES"] = "foo"
+    assert_match "Setting DYLD_INSERT_LIBRARIES",
+      @checks.check_DYLD_vars
+  end
+
+  def test_check_for_symlinked_cellar
+    HOMEBREW_CELLAR.rmtree
+
+    mktmpdir do |path|
+      FileUtils.ln_s path, HOMEBREW_CELLAR
+
+      assert_match path,
+        @checks.check_for_symlinked_cellar
+    end
+
+  ensure
+    HOMEBREW_CELLAR.unlink
+    HOMEBREW_CELLAR.mkpath
+  end
+
+  def test_check_for_autoconf
+    MacOS::Xcode.stubs(:provides_autotools?).returns true
+    mktmpdir do |path|
+      FileUtils.touch "#{path}/autoconf"
```

Comment:
This needs to be made executable to be found by `which` in `check_for_autoconf`.